### PR TITLE
Show "Help Web discovery" in the welcome page for JP

### PR DESCRIPTION
**Pull request by sangwoo108**
This page is intended to be shown only for Japanese users. 
For test purpose, there's a feature flag to force show the page.

* In order to put transition between `ViewTypes`, introduces `useViewTypeTransition` hook.
* Hopefully, with this we don't have to explore all files when adding or removing a node between `ViewType`s.

image

Resolves https://github.com/brave/brave-browser/issues/31545

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### For non JP
* Turn on the `Show Help WDP in Welcome Page` flag
* Open `brave://welcome` page
* Check if a page like captured above appears after import or theme page

### For JP
* Make it sure that the country code is 'JP' - if the country code is 'jp', the additional help page will be opened along with `brave://welcome`
* Remove all user data so that `brave://welcome` page can show up in the first run.
* Check if a page like captured above appears after import or theme page

### CODE REVIEW

File: browser/ui/webui/brave_welcome_ui.cc
Comment (sangwoo108) [inline browser/ui/webui/brave_welcome_ui.cc:1]: I guess this was leftover file.

**Comment by sangwoo108** [comment]
cc: @DJAndries, just a heads up in case we decide to add more metrics.

File: browser/ui/webui/welcome_page/brave_welcome_ui.cc [hunk 1]
```diff
@@ -131,8 +138,10 @@ BraveWelcomeUI::BraveWelcomeUI(content::WebUI* web_ui, const std::string& name)
 
   // Open additional page in Japanese region
   int country_id = country_codes::GetCountryIDFromPrefs(profile->GetPrefs());
+  const bool isJPN =
```
Comment (simonhong) [inline browser/ui/webui/welcome_page/brave_welcome_ui.cc:141]: nit:
```suggestion
  const bool is_jpn =
```

File: components/brave_welcome_ui/components/help-wdp/style.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  a {
+    text-decoration: underline;
+    color: currentColor;
+    font-size: 13px;
+    line-height: 24px;
+    letter-spacing: 0.13px;
+  }
+`
+
+export const ActionBox = styled.div`
```
Comment (simonhong) [inline components/brave_welcome_ui/components/help-wdp/style.ts:77]: It seems `isTertiary` button doesn't have `color` property while `isPrimary` has `white`.
Maybe we want to use `white` for `Maybe later` button?
image

Comment (sangwoo108) [reply components/brave_welcome_ui/components/help-wdp/style.ts:77]: Good call. Set `color` to button.

File: components/definitions/welcome.d.ts [hunk 3]
```diff
@@ -1,3 +1,7 @@
+//  Copyright (c) 2018 The Brave Authors. All rights reserved.
+//  This Source Code Form is subject to the terms of the Mozilla Public
+//  License, v. 2.0. If a copy of the MPL was not distributed with this file,
+//  You can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (simonhong) [inline components/definitions/welcome.d.ts:4]:
```suggestion
//  You can obtain one at https://mozilla.org/MPL/2.0/.

```

File: components/brave_welcome_ui/state/hooks.ts [hunk 4]
```diff
@@ -65,3 +66,67 @@ export function useProfileCount () {
 
 export const shouldPlayAnimations = loadTimeData.getBoolean('hardwareAccelerationEnabledAtStartup') &&
     !window.matchMedia('(prefers-reduced-motion: reduce)').matches
+
```
Comment (simonhong) [inline components/brave_welcome_ui/state/hooks.ts:69]:
```suggestion
```

File: components/definitions/welcome.d.ts [hunk 5]
```diff
@@ -44,5 +48,6 @@ declare namespace Welcome {
     browserProfiles: Array<BrowserProfile>,
     showSearchCard: boolean,
     showRewardsCard: boolean
+    countryString: string
```
Comment (simonhong) [inline components/definitions/welcome.d.ts:51]: Maybe `countryString` is leftover? I think it could be deleted.

File: components/brave_welcome_ui/state/context.ts [hunk 6]
```diff
@@ -16,6 +16,7 @@ interface Store {
   browserProfiles: BrowserProfile[] | undefined
   viewType: ViewType | undefined
   scenes: Scenes | undefined
+  countryString: string | undefined
```
Comment (simonhong) [inline components/brave_welcome_ui/state/context.ts:19]: ditto

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: components/definitions/welcome.d.ts (hunk 5 above)
Comment (sangwoo108) [reply components/definitions/welcome.d.ts:51]: Oh yes, well spotted.

File: components/brave_welcome_ui/state/data-context-provider.tsx [hunk 7]
```diff
@@ -29,32 +38,37 @@ function DataContextProvider (props: DataContextProviderProps) {
...
       // TODO(tali): Handle item based events
 
-      if (status.event === 'ImportStarted' && (profileCountRef.current > 0)) {
-        setViewType(ViewType.ImportInProgress)
+      if (status.event === 'ImportStarted' && profileCountRef.current > 0) {
+        setViewType(importInProgress)
       }
 
       if (status.event === 'ImportEnded') {
+        console.log('BYE')
```
Comment (nullhook) [inline components/brave_welcome_ui/state/data-context-provider.tsx:61]: left over?

**Review by nullhook — CHANGES_REQUESTED** [review]
this looks good

**Review by nullhook — CHANGES_REQUESTED** [review]
starting to look good

File: components/brave_welcome_ui/state/data-context-provider.tsx (hunk 7 above)
Comment (sangwoo108) [reply components/brave_welcome_ui/state/data-context-provider.tsx:61]: Oops, 😄

**Comment by nullhook** [comment]
image

do you think we can reduce the aliasing around the magnifying glass? this should be a design thing. not sure who designed it?

File: components/brave_welcome_ui/assets/brave-search-cube.svg [hunk 8]
```diff
@@ -0,0 +1,79 @@
+<svg width="218" height="182" viewBox="0 0 218 182" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (nullhook) [inline components/brave_welcome_ui/assets/brave-search-cube.svg:1]: did you optimize this svg? https://jakearchibald.github.io/svgomg/

File: components/brave_welcome_ui/state/data-context-provider.tsx [hunk 9]
```diff
@@ -8,53 +8,73 @@ import { addWebUiListener } from 'chrome://resources/js/cr.js'
...
-    setScenes,
     browserProfiles,
     currentSelectedBrowser,
-    viewType,
-    scenes
+    setCurrentSelectedBrowser,
+    currentSelectedBrowserProfiles,
+    scenes,
+    setScenes,
+    countryString: loadTimeData.getString('countryString')
```
Comment (nullhook) [inline components/brave_welcome_ui/state/data-context-provider.tsx:49]: why do we need `countryString` in the context? i can't find it's usage anywhere in the code.

File: components/brave_welcome_ui/state/data-context-provider.tsx [hunk 10]
```diff
@@ -8,53 +8,73 @@ import { addWebUiListener } from 'chrome://resources/js/cr.js'
...
   const { profileCountRef, incrementCount, decrementCount } = useProfileCount()
   const [scenes, setScenes] = React.useState<Scenes | undefined>(undefined)
 
+  const currentSelectedBrowserProfiles = React.useMemo(() => {
+    return browserProfiles?.filter(
+      (profile) => profile.browserType === currentSelectedBrowser
+    )
+  }, [browserProfiles, currentSelectedBrowser])
+
   const store = {
```
Comment (nullhook) [inline components/brave_welcome_ui/state/data-context-provider.tsx:39]: is the type for `store` even being inferred here? if not we can make the `interface Store {}` exportable and explicitly anotate the type here.

Comment (sangwoo108) [reply components/brave_welcome_ui/state/data-context-provider.tsx]: Yeah, that's a good idea. Done - used `Store` interface we had in ../state/context.ts

**Review by simonhong — APPROVED** [review]
++ if @nullhook approves :)

File: components/brave_welcome_ui/components/help-wdp/index.tsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+  const handleReject = () => {
+    setViewType(forward)
+  }
+
+  return (
+    <S.MainBox>
+      <div className='view-header-box'>
+        <div className='view-logo-box'>
+          <img src={cubeImg} />
+        </div>
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply components/brave_welcome_ui/components/help-wdp/index.tsx]: Yeah, removing the logo box div would be better. But it seems we still need `left` as alignment is ignored duto `absolute`. Also I found that the svg's width and height was wrong. That might be the cause of the aliasing.

File: components/brave_welcome_ui/assets/brave-search-cube.svg (hunk 8 above)
Comment (sangwoo108) [reply components/brave_welcome_ui/assets/brave-search-cube.svg]: Thanks! done.

File: components/brave_welcome_ui/components/welcome/index.tsx [hunk 12]
```diff
@@ -71,34 +83,44 @@ function Welcome () {
...
+      <div
+        className={classnames({
+          'view-content': true,
+          'initial': shouldPlayAnimations
+        })}
+      >
+        <div className='view-header-box'>
+          <div className='view-details'>
+            <h1 className='view-title'>{getLocale('braveWelcomeTitle')}</h1>
+            <p className='view-desc'>{getLocale('braveWelcomeDesc')}</p>
```
Comment (nullhook) [inline components/brave_welcome_ui/components/welcome/index.tsx:98]: There are changes in this file related to Prettier rules that aren't directly related to the PR. We suggest applying those style changes across the directory in a separate, isolated PR. Could you please revert these changes.

File: components/brave_welcome_ui/components/select-profile/index.tsx [hunk 13]
```diff
@@ -23,17 +28,17 @@ interface ProfileItemProps {
...
 function ProfileItem (props: ProfileItemProps) {
   return (
-    <div className="item-box">
-      <label className="item-grid">
-        <div className="item-info">
-          <div className="avatar">
+    <div className='item-box'>
+      <label className='item-grid'>
+        <div className='item-info'>
+          <div className='avatar'>
```
Comment (nullhook) [inline components/brave_welcome_ui/components/select-profile/index.tsx:34]: ditto

File: components/brave_welcome_ui/components/select-theme/index.tsx [hunk 14]
```diff
@@ -43,58 +46,71 @@ function ThemeModeItem (props: ThemeModeItemProps) {
...
+              fill='#4C54D2'
+            />
           </svg>
         )}
       </i>
-      <div className="logo-box">
+      <div className='logo-box'>
         <div className={logoBoxClass} />
-        <p className="theme-name">{props.title}</p>
+        <p className='theme-name'>{props.title}</p>
```
Comment (nullhook) [inline components/brave_welcome_ui/components/select-theme/index.tsx:69]: ditto

File: components/brave_welcome_ui/components/select-theme/index.tsx [hunk 15]
```diff
@@ -43,58 +46,71 @@ function ThemeModeItem (props: ThemeModeItemProps) {
...
-        <div className="view-header-actions">
-        </div>
-        <div className="view-details">
-          <h1 className="view-title">{getLocale('braveWelcomeSelectThemeLabel')}</h1>
+      <div className='view-header-box'>
+        <div className='view-header-actions'></div>
+        <div className='view-details'>
+          <h1 className='view-title'>
+            {getLocale('braveWelcomeSelectThemeLabel')}
+          </h1>
```
Comment (nullhook) [inline components/brave_welcome_ui/components/select-theme/index.tsx:103]: ditto

**Review by nullhook — CHANGES_REQUESTED** [review]

**Review by nullhook — APPROVED** [review]
nice work

**Comment by minkyeongshin** [comment]
> > do you think we can reduce the aliasing around the magnifying glass? this should be a design thing. not sure who designed it?
> 
> As it's SVG, I think the aliasing isn't from the asset itself. I'll try running svgo as you suggeseted. possibly, it could help by removing some path points.



> image
> do you think we can reduce the aliasing around the magnifying glass? this should be a design thing. not sure who designed it?



> image
> do you think we can reduce the aliasing around the magnifying glass? this should be a design thing. not sure who designed it?

The aliasing around the magnifying glass is caused by the background shadow. I've removed the background shadow, but it appears that there isn't enough contrast. If we don't need to use gradient colors, using a solid color seems acceptable.

Here's the link 
https://www.figma.com/file/MFHpcXJMg2RUQpUJMAZZ4R/Desktop-Onboarding?type=design&node-id=6164%3A15444&mode=design&t=kyWbDrtWYm9JvP5V-1

**Comment by sangwoo108** [comment]
@minkyeongshin Thanks for checking. Do you want me to replace the asset?

**Comment by rebron** [comment]
@sangwoo108 You can use the new image from @minkyeongshin

**Comment by anthonypkeane** [comment]
Thanks @sangwoo108 Will the translations from https://github.com/brave/brave-browser/issues/31545#issuecomment-1859423581 be used?

**Comment by sangwoo108** [comment]
> Thanks @sangwoo108 Will the translations from [brave/brave-browser#31545 (comment)](https://github.com/brave/brave-browser/issues/31545#issuecomment-1859423581) be used?

Yes, it was included!
